[quant][graphmode][jit][api] Expose preserved_attrs from finalize to convert_jit

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#44490 [quant][graphmode][jit][api] Expose preserved_attrs from finalize to convert_jit**

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23631142](https://our.internmc.facebook.com/intern/diff/D23631142)

jerryzh168 committed:
[quant][graphmode][jit][api] Expose preserved_attrs from finalize to convert_jit

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][jit][api] Expose preserved_attrs from finalize to convert_jit"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23631142](https://our.internmc.facebook.com/intern/diff/D23631142)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][jit][api] Expose preserved_attrs from finalize to convert_jit"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23631142](https://our.internmc.facebook.com/intern/diff/D23631142)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][jit][api] Expose preserved_attrs from finalize to convert_jit"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23631142](https://our.internmc.facebook.com/intern/diff/D23631142)

[ghstack-poisoned]

z-a-f commented on torch/csrc/jit/passes/quantization/finalize.cpp:
```diff
@@ -73,12 +73,14 @@ void FoldQuantizedPrepackingOps(Module& module) {
 }
 
-Module Finalize(Module& module, QuantType quant_type) {
+Module Finalize(Module& module,
+                QuantType quant_type,
+                const std::vector<std::string>& preserved_attrs) {
```
nit: are these subject to BC?

jerryzh168 replied:
you mean does this affect BC? I don't think so, these is an internal function.

for user function convert_jit this is an optional argument, so it won't affect BC either.

z-a-f reviewed (approved):
lgtm

jerryzh168 committed:
Update on "[quant][graphmode][jit][api] Expose preserved_attrs from finalize to convert_jit"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23631142](https://our.internmc.facebook.com/intern/diff/D23631142)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][jit][api] Expose preserved_attrs from finalize to convert_jit"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23631142](https://our.internmc.facebook.com/intern/diff/D23631142)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][jit][api] Expose preserved_attrs from finalize to convert_jit"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23631142](https://our.internmc.facebook.com/intern/diff/D23631142)

[ghstack-poisoned]
